Cloudinary/Cloudinary.cs

```diff
@@ -11,960 +12,886 @@ namespace CloudinaryDotNet
-        /// </exception>
-        public RawUploadResult UploadLargeRaw(BasicRawUploadParams parameters, int bufferSize = 20 * 1024 * 1024)
-        {
-            if (parameters is RawUploadParams)
-                throw new ArgumentException("Please use BasicRawUploadParams class for large raw file uploading!");
-
-            parameters.Check();
-
-            if (parameters.File.IsRemote)
-                throw new ArgumentException("The UploadLargeRaw method is intended to be used for large local file uploading and can't be used for remote file uploading!");
-            return UploadLarge(parameters, bufferSize, true) as RawUploadResult;
-        }
-
-        private string RandomPublicId()
-        {
-            byte[] buffer = new byte[8];
-            m_random.NextBytes(buffer);
-            return string.Concat(buffer.Select(x => x.ToString("X2")).ToArray());
-        }
-
-        public RawUploadResult UploadLarge(RawUploadParams parameters, int bufferSize = 20 * 1024 * 1024)
+    public partial class Cloudinary : CloudinaryShared
+    {
+        /// <summary>
+        /// Default parameterless constructor.
+        /// Assumes that environment variable CLOUDINARY_URL is set.
+        /// </summary>
+        public Cloudinary()
+        {
+            m_api = new Api();
+        }
+
+        /// <summary>
+        /// Parameterized constructor
+        /// </summary>
+        /// <param name="cloudinaryUrl">Cloudinary URL</param>
+        public Cloudinary(string cloudinaryUrl)
+        {
+            m_api = new Api(cloudinaryUrl);
+        }
+
+        /// <summary>
+        /// Parameterized constructor
+        /// </summary>
+        /// <param name="account">Cloudinary account</param>
+        public Cloudinary(Account account)
+        {
+            m_api = new Api(account);
+        }
+
+        /// <summary>
+        /// API object that used by this instance
+        /// </summary>
+        public Api Api
+        {
+            get { return (Api)m_api; }
+        }
+
+        /// <summary>
+        /// This method can be used to force refresh facebook and twitter profile pictures. The response of this method includes the image's version. Use this version to bypass previously cached CDN copies.
+        /// Also it can be used to generate transformed versions of an uploaded image. This is useful when Strict Transformations are allowed for your account and you wish to create custom derived images for already uploaded images. 
+        /// </summary>
+        /// <param name="parameters">The parameters.</param>
+        /// <returns></returns>
+        public ExplicitResult Explicit(ExplicitParams parameters)
+        {
+            string uri = m_api.ApiUrlImgUpV.Action("explicit").BuildUrl();
+
+            using (HttpWebResponse response =((Api)m_api).Call(HttpMethod.POST, uri, parameters.ToParamsDictionary(), null))

```

Comment:
yes. It will be great. as for when...I think we are going to have other changes, so ...then...